packages/frontend/src/page/components/wallet-position/actions/index.tsx

```diff
@@ -1,61 +1,142 @@
+            redeemableRewards.length > 0 &&
+            redeemableRewards.some((reward) => reward.isPositive());
+        setRedeemable(hasSomeRedeemableReward);
 
-        for (const redeemable of redeemableRewards) {
-            if (!redeemable.isZero()) {
-                setRedeemable(true);
-                return;
-            }
+        if (
+            kpiToken.expired &&
+            !hasSomeRedeemableReward &&
+            kpiTokenBalance?.gt(0)
+        ) {
+            setBurnable(true);
         }
-        setRedeemable(false);
-    }, [redeemableRewards]);
+    }, [kpiToken.expired, kpiTokenBalance, redeemableRewards]);
+
+    useEffect(() => {
+        if (kpiToken.expired) {
+            setText(
+                kpiTokenBalance && kpiTokenBalance.gt(0)
+                    ? t("position.status.expired.withBalance", {
+                          amount: formatTokenAmount(kpiTokenBalance, false),
+                          symbol: kpiTokenBalance.currency.symbol,
+                      })
+                    : t("position.status.expired.noBalance")
+            );
+        } else {
+            if (kpiToken.finalized) {
+                if (kpiTokenBalance && kpiTokenBalance.gt(0)) {
+                    if (burnable)
+                        setText(
+                            t(
+                                "position.status.finalized.withBalance.burnable",
+                                {
+                                    amount: formatTokenAmount(
+                                        kpiTokenBalance,
+                                        false
+                                    ),
+                                    symbol: kpiTokenBalance.currency.symbol,
+                                }
+                            )
+                        );
+                    else if (redeemable)
+                        setText(
+                            t(
+                                "position.status.finalized.withBalance.redeemable",
+                                {
+                                    amount: formatTokenAmount(
+                                        kpiTokenBalance,
+                                        false
+                                    ),
+                                    symbol: kpiTokenBalance.currency.symbol,
+                                }
+                            )
+                        );
+                } else setText(t("position.status.finalized.noBalance"));
+            } else setText(t("position.status.notFinalized"));
+        }
+    }, [
+        burnable,
+        kpiToken.expired,
+        kpiToken.finalized,
+        kpiTokenBalance,
+        redeemable,
+        t,
+    ]);
 
     const handleClick = useCallback(async () => {
         if (!writeAsync) return;
         setLoading(true);
         try {
-            await writeAsync();
+            const tx = await writeAsync();
+            await tx.wait();
```

Comment:
Should we add the transaction on the account popover? Using the `onTx` function.

Reply:
You're right, I'll add that feature.